### PR TITLE
Add module web_custom_modifier

### PR DESCRIPTION
**Pull request by ddufresne**
https://www.numigi.com/web#id=16243&action=292&active_id=595&model=project.task&view_type=form&menu_id=200

### CODE REVIEW

File: web_custom_modifier/models/ir_ui_view.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
+from lxml import etree
+from typing import List
+from odoo import api, fields, models
+from odoo.osv.orm import transfer_node_to_modifiers, transfer_modifiers_to_node
+
+
+def _add_custom_modifier_to_view_tree(modifier: dict, tree: etree._Element):
```
Comment (ddufresne) [inline web_custom_modifier/models/ir_ui_view.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=4709570)

File: web_custom_modifier/tests/test_view_rendering.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
+from ddt import data, ddt
+from lxml import etree
```
Comment (ddufresne) [inline web_custom_modifier/tests/test_view_rendering.py]: Codacy Issue found: [Using etree to parse untrusted XML data is known to be vulnerable to XML attacks. Replace etree with the equivalent defusedxml package.](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=4709570)

File: web_custom_modifier/models/ir_ui_view.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
+from lxml import etree
+from typing import List
+from odoo import api, fields, models
```
Comment (ddufresne) [inline web_custom_modifier/models/ir_ui_view.py]: Codacy Issue found: [Unused fields imported from odoo](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=4709570)

File: web_custom_modifier/models/ir_ui_view.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    """Add custom modifiers to the given view architecture.
+
+    :param arch: the architecture to extend.
+    :param modifiers: a list of custom modifiers to apply.
+    :return: the view architecture with custom modifiers.
+    """
+    if not modifiers:
+        return arch
+
+    tree = etree.fromstring(arch)
```
Comment (ddufresne) [inline web_custom_modifier/models/ir_ui_view.py]: Codacy Issue found: [Using lxml.etree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree.fromstring with its defusedxml equivalent function.](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=4709570)

File: web_custom_modifier/tests/test_view_rendering.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        cls.street_modifier = cls.env['web.custom.modifier'].create({
+            'model_ids': [(4, cls.env.ref('base.model_res_partner').id)],
+            'type_': 'xpath',
+            'reference': cls.xpath,
+            'modifier': 'invisible',
+        })
+
+    def _get_rendered_view_tree(self):
+        arch = self.env['res.partner'].fields_view_get(view_id=self.view.id)['arch']
+        return etree.fromstring(arch)
```
Comment (ddufresne) [inline web_custom_modifier/tests/test_view_rendering.py]: Codacy Issue found: [Using lxml.etree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree.fromstring with its defusedxml equivalent function.](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=4709570)

File: web_custom_modifier/models/ir_ui_view.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
+from lxml import etree
```
Comment (ddufresne) [inline web_custom_modifier/models/ir_ui_view.py]: Codacy Issue found: [Using etree to parse untrusted XML data is known to be vulnerable to XML attacks. Replace etree with the equivalent defusedxml package.](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=4709570)

File: web_custom_modifier/models/__init__.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from . import (
```
Comment (ddufresne) [inline web_custom_modifier/models/__init__.py]: Codacy Issue found: ['.ir_ui_view' imported but unused](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=4709570)

File: web_custom_modifier/models/ir_ui_view.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
```
Comment (ddufresne) [inline web_custom_modifier/models/ir_ui_view.py]: Codacy Issue found: [Unused import json](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=4709570)

**Review by foutoucour — APPROVED** [review]

**Pull request merged**